app/policies/provider_policy.rb

```diff
@@ -0,0 +1,9 @@
+class ProviderPolicy < ApplicationPolicy
+  class Scope < ApplicationPolicy::Scope
+    def resolve
+      return scope if user.support_user?
+
+      scope.where(id: user.current_organisation.partner_providers.ids)
```

Comment:
Let's use a subquery for these instead of causing extra queries:

```suggestion
      scope.where(id: user.current_organisation.partner_providers.select(:id))
```